fs/graph/graph.go

```diff
@@ -38,7 +39,17 @@ func Request(resource string, auth *Auth, method string, content io.Reader) ([]b
 
 	auth.Refresh()
 
-	client := &http.Client{Timeout: 15 * time.Second}
+	client := &http.Client{
```

Comment:
I know this is a weird request, but can you just increase the existing timeout and not add in the fine-grained configuration? I'd prefer to avoid the extra configuration unless there is a reason to do so. ([Also the `ResponseHeaderTimeout` has actually been the problem in the past](https://github.com/jstaf/onedriver/issues/98), and this actually decreases that timeout from 15 to 10 seconds.)

Maybe just change the global timeout to 30s?

Reply:
I understand your request. I tested to change the original timeout to 600 seconds to ensure the cause are the timeout. It works but if the value are 30 seconds if the download exceed this, the issue will occurred again.
After search an explanation about timeout of http client, I found [this](https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/)

I based my modification on that and I set more fine timeout to avoid a uncatchable timeout on body reading (We receive headers so all are OK). And with values in my commit the timeout on header reading occur 50 seconds if all time are used or ~10s if connect is established quickly after client.Do call instead of 15 seconds for all request's steps as detailed in above link.

I don't know if these timeout values are perfect and will not cause another issue, but we can change these if necessary to be more pragmatic.

I tried to download 5GB file without issue and the sha256sum are same.

Reply:
Honestly, I think the fix in this case is implementing the "partial range downloads" described here: https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_get_content?view=odsp-graph-online#partial-range-downloads. Ironically, this technique gets used for uploads, but not downloads right now.

I also found the Cloudflare blog while looking for where you got those settings, and I don't think it was a recommendation there, it was more of a "here is how you set these different timeouts". Right now I am happy to just increase the global timeout, even if this is just a band-aid solution until the "partial range downloads" stuff is implemented.

Reply:
The partial download will reduce error but not prevent it. If your connection is too slow and you client timeout on response body reading, you will get partial content of your partial download. And finally you will have a corrupted file.
I don't found how to catch a timeout on body reading and in my tests the timeout is quiet.

For the Cloudflare blog, you are right, it's not recommendation and as I said, it's explanation.
If increasing the global timeout is good for you, you can close this request.

Thank you.